pages/api/links/[id]/index.ts

```diff
@@ -128,6 +126,38 @@ export default async function handle(
     }
 
     return res.status(200).json(updatedLink);
+  } else if (req.method == "DELETE") {
+    const session = await getServerSession(req, res, authOptions);
+    if (!session) {
+      return res.status(401).end("Unauthorized");
+    }
+
+    const { id } = req.query as { id: string };
+
+    try{
+      const linkToBeDeleted = await prisma.link.findUnique({
+        where: {
+          id: id,
+        }
+      });
```

Comment:
I think we should add a way to determine that the user requesting to delete the link is actually authorized to delete the link. 

I like that you put this method in here. 
This allows us to compare the ownerId with the session userId, if you include the `document: { ownerId: true }`  on this function.